internal/graph/memory.go

```diff
@@ -116,19 +117,54 @@ func (m *Memory) Predecessors(_ context.Context, node ocispec.Descriptor) ([]oci
 	return res, nil
 }
 
+// RemoveFromIndex removes the node from its predecessors and successors.
+func (m *Memory) RemoveFromIndex(ctx context.Context, node ocispec.Descriptor) error {
```

Comment:
There is a race condition where the same node is indexed and removed at the same time.

Reply:
For example, what if another goroutine calls `Index()` when the current goroutine just about to hit the line 132?